agentlib/modules/controller/controller.py

```diff
@@ -123,4 +123,4 @@ def _siso_callback(self, inp: AgentVariable, name: str):
         else:
             out_name = self.config.output.name
             self.logger.debug("Sending output %s=%s", out_name, out_val)
-            self.set(name=out_name, value=out_val, timestamp=inp.timestamp)
```

Comment:
This one is a bit weird. Now that I understand the environment better, I think maybe this inp.timestamp is correct for real time cases. But there should be an even better solution, which I will try to find in #18